### PR TITLE
Library and sample app has been migrated to kotlin

### PR DESCRIPTION
**Pull request by anoop44**

### CODE REVIEW

**Comment by yshrsmz** [comment]
Hi, sorry for being late at the review!

I'd like to know the reasoning behind this PR before proceeding to the actual review, as simply migrating to Kotlin does not provide any value to users.

Maybe we can add some extension functions to make our life easier, but that can be a different module.

**Comment by anoop44** [comment]
Hey @yshrsmz ,
You are right, this will not provide any new feature or value to users. But going forward kotlin might help in adding new features or providing some extension. For now its more like going with the trend of Kotlin over Java

**Comment by yshrsmz** [comment]
Thanks a lot, let's migrate to Kotlin then 👍

File: keyboardvisibilityevent/src/main/java/net/yslibrary/android/keyboardvisibilityevent/SimpleUnregistrar.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+/**
+ * @author Anoop S S
+ * anoopvvs@gmail.com
+ * on 28/02/2017
+ */
+
+class SimpleUnregistrar internal constructor(activity: Activity, globalLayoutListener: ViewTreeObserver.OnGlobalLayoutListener) : Unregistrar {
+
+    private val mActivityWeakReference: WeakReference<Activity> = WeakReference(activity)
```
[the remark this answers is not in the document]
Comment (anoop44) [reply keyboardvisibilityevent/src/main/java/net/yslibrary/android/keyboardvisibilityevent/SimpleUnregistrar.kt]: Done

File: keyboardvisibilityevent/src/main/java/net/yslibrary/android/keyboardvisibilityevent/util/UIUtil.kt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import android.widget.EditText
+
+/**
+ * Created by yshrsmz on 15/03/17.
+ */
+class UIUtil
+/**
+ * Suppresses instantiation
+ */
+private constructor() {
```
[the remark this answers is not in the document]
Comment (anoop44) [reply keyboardvisibilityevent/src/main/java/net/yslibrary/android/keyboardvisibilityevent/util/UIUtil.kt]: Done

File: keyboardvisibilityevent/src/main/java/net/yslibrary/android/keyboardvisibilityevent/KeyboardVisibilityEvent.kt [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+object  KeyboardVisibilityEvent {
+
+    /**
+     * Set keyboard visibility change event listener.
+     * This automatically remove registered event listener when the Activity is destroyed
+     *
+     * @param activity Activity
+     * @param listener KeyboardVisibilityEventListener
+     */
+    fun setEventListener(activity: Activity,
```
[the remark this answers is not in the document]
Comment (anoop44) [reply keyboardvisibilityevent/src/main/java/net/yslibrary/android/keyboardvisibilityevent/KeyboardVisibilityEvent.kt]: Done

**Pull request merged**